### PR TITLE
Add IsHardened in launch spec

### PR DESCRIPTION
**Pull request by jkl73**
~~* COS eventlog now can measure to a resettable PCR (in the debug image)~~
~~* Debug PCR will reset every time before launcher run in the debug image~~
* Added a kernel cmdline arg confidential-space.hardened=true to indicate hardened image
* Change LauncherSpec to LaunchSpec
* Add cgo flag to silence deprecate warning in openssl for ms-tpm-20-ref

Signed-off-by: Jiankun Lu <jiankun@google.com>

### CODE REVIEW

**Comment by jkl73** [comment]
I'm not sure why git didn't recognize the move `launcher_spec_test.go -> launch_spec_test.go``, the only meaningful change in this file is the new function at the end `TestIsHardened`

**Comment by josephlr** [comment]
We will need to see what we do here after #246 is merged.

My preference would probably be to just have PCR 13 used unconditionally, but have the server side be capable of understanding multiple container launches.

> I'm not sure why git didn't recognize the move `launcher_spec_test.go -> launch_spec_test.go`, the only meaningful change in this file is the new function at the end `TestIsHardened`

For future reference, you can get Git to pay attention to this stuff by having one commit that just moves the file (with `git mv`) and another that actually makes the change.

**Comment by jkl73** [comment]
> We will need to see what we do here after #246 is merged.
> 
> My preference would probably be to just have PCR 13 used unconditionally, but have the server side be capable of understanding multiple container launches.
> 
> > I'm not sure why git didn't recognize the move `launcher_spec_test.go -> launch_spec_test.go`, the only meaningful change in this file is the new function at the end `TestIsHardened`
> 
> For future reference, you can get Git to pay attention to this stuff by having one commit that just moves the file (with `git mv`) and another that actually makes the change.

Updated, removed resettable PCR change. I split the commit so now I think git can recognize the file rename.

**Comment by josephlr** [comment]
Maybe I'm missing something, but what is the future purpose of the IsHardened value in the launch spec? Is this so we can entirely control the machine restart behavior inside the container launcher? or something else?

File: launcher/spec/launch_spec.go [hunk 1]
```diff
@@ -121,30 +123,55 @@ func getRegion(client *metadata.Client) (string, error) {
...
 }
+
+func isHardened(kernelCmd string) bool {
+	args := strings.Fields(kernelCmd)
+
+	for _, arg := range args {
+		if strings.HasPrefix(arg, "confidential-space.hardened=") {
+			return strings.HasSuffix(arg, "=true")
+		}
+	}
```
Comment (josephlr) [inline launcher/spec/launch_spec.go:167]: Can this be simplified to:

```go
for _, arg := range strings.Fields(kernelCmd) {
  if arg == "confidential-space.hardened=true" {
    return true
  }
}
return false
```

File: simulator/internal/internal_cgo.go [hunk 2]
```diff
@@ -19,6 +19,8 @@ package internal
 // // Silence known warnings from the reference code and CGO code.
 // #cgo CFLAGS: -Wno-missing-braces -Wno-empty-body -Wno-unused-variable -Wno-uninitialized
 // // Link against the system OpenSSL
+// // Silence openssl deprecation warnings for ms-tpm-20-ref
+// #cgo CFLAGS: -Wno-deprecated-declarations
```
Comment (josephlr) [inline simulator/internal/internal_cgo.go:23]: This should go above the comment about linking against openssl

**Review by josephlr — APPROVED** [review]

File: launcher/spec/launch_spec.go (hunk 1 above)
Comment (jkl73) [reply launcher/spec/launch_spec.go:167]: done

File: simulator/internal/internal_cgo.go (hunk 2 above)
Comment (jkl73) [reply simulator/internal/internal_cgo.go:23]: done

**Pull request merged**